packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -678,6 +681,15 @@ function getFunctionName(node) {
       //
       // class {useHook = () => {}}
       // class {useHook() {}}
+    } else if (
+      node.parent.type === 'JSXExpressionContainer' &&
+      node.parent.parent.type === 'JSXAttribute' &&
+      node.parent.parent.name.type === 'JSXIdentifier'
```

Comment:
I think this is overly defensive in todays JSX. But just in case expressioncontainers will ever be able to appear elsewhere and JSXAttributes might be something other than an identifier (e.g. computed 😉 )